apps/server/lib/lexical/server/project/progress/state.ex

```diff
@@ -0,0 +1,88 @@
+defmodule Lexical.Server.Project.Progress.State do
+  alias Lexical.Project
+  alias Lexical.Protocol.Notifications
+  alias Lexical.Server.Project.Progress.Value
+  alias Lexical.Server.Transport
+
+  require Logger
+
+  defstruct project: nil, token_by_label: %{}, progress: nil
```

Comment:
I think I said this before, but why not just store the values here? This seems easier than doing the token by label thing. 
Presumably you can have multiple progresses being updated at the same time, right?